app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,47 @@
+class GroupsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_group, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @groups = Group.all
+  end
+
+  def new
+    @group = Group.new
+  end
+
+  def edit
+  end
+
+  def create
+    @group = Group.new(group_params)
+
+    if @group.save
+      redirect_to groups_path, notice: 'Group was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @group.update(group_params)
+      redirect_to groups_path, notice: 'Group was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
```

Comment:
We may want to allow this only for Admin users or potentially even disable it entirely. This is good for now but perhaps we can create an issue to address that part.